util/src/main/scala/org/ensime/util/path.scala

```diff
@@ -0,0 +1,57 @@
+// Copyright: 2010 - 2016 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.util
+
+import java.nio.charset.Charset
+import java.nio.file.{ Files, Path }
+
+/**
+ * Idiomatic scala methods for Path. One must not assume that Path is
+ * a File, see `ensimefile` for more appropriate functionality.
+ *
+ * (Maybe this should be private within the ensimefile classes)
+ */
+package object path {
+
+  implicit class RichPath(val path: Path) extends AnyVal {
+    // PathMatcher is too complex, use http://stackoverflow.com/questions/20531247
+    def isJava: Boolean = path.toString.toLowerCase.endsWith(".java")
+    def isScala: Boolean = path.toString.toLowerCase.endsWith(".scala")
+    def exists(): Boolean = Files.exists(path)
+    def readBytes(): Array[Byte] = Files.readAllBytes(path)
+    def readString()(implicit cs: Charset): String = new String(readBytes(), cs)
+
+    def /(child: String): Path = path.resolve(child)
+    def canon: Path = ???
+    def isFile: Boolean = path.toFile.isFile
+
+    /*
```

Comment:
yeah, it's a reminder about how to use the NIO API, we've to come back to this for the task to remove VFS.